src/CAN/can_mapping.c

```diff
@@ -47,7 +49,30 @@ float canmapping_extract_value(uint64_t raw_data, const CANMapping *mapping)
                         break;
                 }
         }
-        return (float)raw_value;
+
+        /* convert type */
+        float extracted_value = 0;
+        switch (mapping->type) {
+        	case CANMappingType_IEEE754:
+        		extracted_value = *((float*)&raw_value);
+        		break;
+        	case CANMappingType_signed:
+        		if (mapping->length <= 8) {
+        			extracted_value = *((int8_t*)&raw_value);
+        		}
+        		else if (mapping->length <= 16){
+        			extracted_value = *((int16_t*)&raw_value);
+        		}
+        		else {
+        			extracted_value = *((int32_t*)&raw_value);
+        		}
+        		break;
+        	case CANMappingType_unsigned:
+        	default:
```

Comment:
This seems dubious.  Since this is an enum you should not have a default case here.  You want to handle it this way because if you add a new enum in the future you want the compiler to tell you that you have forgotten a case in a case statement like this.  This is a benefit of using enums with gcc.  So get rid of the default and handle all specific types.